src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/CacheAnnotationsMonologInjectorPass.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\Cache\Adapter\AbstractAdapter;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+final class CacheAnnotationsMonologInjectorPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!($container->hasDefinition('cache.annotations') || $container->hasAlias('cache.annotations'))) {
```

Comment:
The less parenthesis and `OR` present in a condition, the easier it is to understand. Especially in guard clauses you want it to be as specific as possible with `AND`:
`if x AND y AND z; then return`